stdlib/LinearAlgebra/src/LinearAlgebra.jl

```diff
@@ -424,30 +425,69 @@ end
 
 
 function versioninfo(io::IO=stdout)
-    if Base.libblas_name == "libopenblas" || BLAS.vendor() === :openblas || BLAS.vendor() === :openblas64
-        openblas_config = BLAS.openblas_get_config()
-        println(io, "BLAS: libopenblas (", openblas_config, ")")
+    config = BLAS.get_config()
+    println(io, "BLAS: $(BLAS.libblastrampoline) ($(join(string.(config.build_flags), ", ")))")
+    for lib in config.loaded_libs
+        println(io, " --> $(lib.libname) ($(uppercase(string(lib.interface))))")
+    end
+    return nothing
+end
+
+function find_blas_library(name)
+    shlib_ext = string(".", Libdl.dlext)
+    if !endswith(name, shlib_ext)
+        name_ext = string(name, shlib_ext)
+    end
+
+    # On windows, we look in `bin` and never in `lib`
+    @static if Sys.iswindows()
+        path = joinpath(Sys.BINDIR, name_ext)
+        if isfile(path)
+            return path
+        end
     else
-        println(io, "BLAS: ",Base.libblas_name)
+        # On other platforms, we check `lib/julia` first, and if that doesn't exist, `lib`.
+        path = joinpath(Sys.BINDIR, Base.LIBDIR, "julia", name_ext)
+        if isfile(path)
+            return path
+        end
+
+        path = joinpath(Sys.BINDIR, Base.LIBDIR, name_ext)
+        if isfile(path)
+            return path
+        end
     end
-    println(io, "LAPACK: ",Base.liblapack_name)
+
+    # If we can't find it by absolute path, we'll try just passing this straight through to `dlopen()`
+    return name
+end
+
+function get_blas_lapack_path()
+    blas_path = find_blas_library(Base.liblapack_name)
+    lapack_path = find_blas_library(Base.liblapack_name)
+    return (blas_path, lapack_path)
 end
 
 function __init__()
+    # Eventually, this will be provided via libblastrampoline_jll
+    libblastrampoline_handle = Libdl.dlopen(BLAS.libblastrampoline)
+
     try
-        BLAS.check()
-        if BLAS.vendor() === :mkl
-            ccall((:MKL_Set_Interface_Layer, Base.libblas_name), Cvoid, (Cint,), USE_BLAS64 ? 1 : 0)
+        libblas_path = find_blas_library(Base.libblas_name)
+        liblapack_path = find_blas_library(Base.liblapack_name)
+        BLAS.lbt_forward(libblas_path; clear=true)
+        if liblapack_path != libblas_path
+            BLAS.lbt_forward(liblapack_path)
         end
+        BLAS.check()
         Threads.resize_nthreads!(Abuf)
         Threads.resize_nthreads!(Bbuf)
         Threads.resize_nthreads!(Cbuf)
-    catch ex
-        Base.showerror_nostdio(ex,
-            "WARNING: Error during initialization of module LinearAlgebra")
-    end
-    # register a hook to disable BLAS threading
-    Base.at_disable_library_threading(() -> BLAS.set_num_threads(1))
+     catch ex
+        Base.showerror_nostdio(ex, "WARNING: Error during initialization of module LinearAlgebra")
+     end
+     # register a hook to disable BLAS threading
+     Base.at_disable_library_threading(() -> BLAS.set_num_threads(1))
```

Comment:
Seems to be overly indented
```suggestion
    catch ex
        Base.showerror_nostdio(ex, "WARNING: Error during initialization of module LinearAlgebra")
    end
    # register a hook to disable BLAS threading
    Base.at_disable_library_threading(() -> BLAS.set_num_threads(1))
```